src/Mapper/ORM/DoctrineORMMapper.php

```diff
@@ -66,13 +64,25 @@ public function getSubscribedEvents(): array
         ];
     }
 
-    public function addAssociation(string $class, string $field, OptionsBuilder $options): void
+    /**
+     * @param array $options
+     */
+    public function addAssociation(string $class, string $type, $options): void
     {
+        // NEXT_MAJOR: Move array check to method signature
+        if (!\is_array($options)) {
+            throw new \TypeError(sprintf(
```

Comment:
This is a BC break :( 

You could add a deprecation error to keep BC

Reply:
I agree, but I also think a new exception like this could be considered as BC compliant if the previous code throws a similar exception sooner or later because the unsupported type, but I don't know if it is the case.

Reply:
This type used was changed.
From
```
public function addAssociation(string $class, string $field, OptionsBuilder $options): void
```
To
```
public function addAssociation(string $class, string $field, array $options): void
```

Reply:
You're right. Given the signature change I was assuming the `OptionsBuilder` declaration was added in a yet unreleased version, but I see that's not the case:

https://github.com/sonata-project/sonata-doctrine-extensions/blob/9f39109af81561f8175a3bcece454724a8d9e053/src/Mapper/ORM/DoctrineORMMapper.php#L69

This is really a BC break.
Thanks for pointing.

Reply:
Why is this a BC break? this Pull request is about fixing a bug because of this type hint.

Reply:
The function alone could still be used by someone with an OptionBuilder as a third argument.

You just have to trigger a deprecation like this https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Admin/AbstractAdmin.php#L1079 in order to be BC.